src/bucketizers/timeBucketTree.ts

```diff
@@ -0,0 +1,377 @@
+import { BasicLensM, Cont } from "rdf-lens";
+import { AddRelation, Bucketizer } from ".";
+import { Bucket, Record } from "../utils";
+import { Term } from "@rdfjs/types";
+import { TREE, XSD } from "@treecg/types";
+import { DataFactory } from "n3";
+
+const { literal, namedNode } = DataFactory;
+
+export type Level =
+    | "year"
+    | "month"
+    | "day-of-month"
+    | "hour"
+    | "minute"
+    | "second"
+    | "millisecond";
+
+export type TimeBucket = {
+    ranges: Level[];
+    amount: number;
+};
+
+const months = [
+    "january",
+    "february",
+    "march",
+    "april",
+    "may",
+    "june",
+    "july",
+    "august",
+    "september",
+    "october",
+    "november",
+    "december",
+];
+
+const levelToValue: { [key in Level]: (date: Date) => string } = {
+    year: (date) => date.getUTCFullYear() + "",
+    month: (date) => months[date.getUTCMonth()],
```

Comment:
```suggestion
    month: (date) => date.toLocaleString('en-US', { month: 'long' }).toLowerCase(),
```
Can't we use something like this instead?

Reply:
Hm seems way slower for no benefit imo

Reply:
Cleaner code, but is it actually that much slower?

Reply:
I'll do a micro bench, give me sec

Reply:
```
getMonth
├─formatting x 23,095 ops/sec @ 43μs/op ± 8.73% (min: 32μs, max: 5ms)
└─indexing x 4,016,064 ops/sec @ 249ns/op ± 6.80% (min: 138ns, max: 28μs)
```

Reply:
Certainly worst case is not really acceptable tbh

Reply:
But you give a good point, the code should be more readable
And I should do some more testing as to the performance as dates are not the fastest
( I think this is why my pipeline is currently so slow )